internal/gengapic/gengapic_test.go

```diff
@@ -378,6 +349,89 @@ methods:
 	}
 }
 
+func TestContainsDeprecated(t *testing.T) {
+	tests := []struct {
+		in   string
+		want bool
+	}{
+		{
+			in:   "",
+			want: false,
+		},
+		{
+			in:   "This comment does not contain the prefix Deprecated.\nNot even on this line.",
+			want: false,
+		},
+		{
+			in:   "Deprecated: this is a proper deprecation notice.",
+			want: true,
+		},
+		{
+			in:   "This is a comment that includes a deprecation notice.\nDeprecated: This is a proper deprecation notice.",
+			want: true,
+		},
+		{
+			in:   "This is not a properly formatted Deprecated: notice.\nNeither is this one - Deprecated:",
+			want: false,
+		},
+	}
+
+	for _, tst := range tests {
+		if diff := cmp.Diff(containsDeprecated(tst.in), tst.want); diff != "" {
+			t.Errorf("comment() got(-),want(+):\n%s", diff)
+		}
+	}
+}
+
+func TestMethodDoc(t *testing.T) {
+	m := &descriptor.MethodDescriptorProto{
+		Name: proto.String("MyMethod"),
+	}
+
+	g := generator{
+		comments: make(map[proto.Message]string),
+	}
+
+	for _, tst := range []struct {
+		in, want   string
+		deprecated bool
+	}{
+		{
+			in:   "",
+			want: "",
+		},
+		{
+			in:   "Does stuff.\n It also does other stuffs.",
+			want: "// MyMethod does stuff.\n// It also does other stuffs.\n",
+		},
+		{
+			in:         "This is deprecated.\n It does not have a proper comment.",
+			want:       "// This is deprecated.\n// It does not have a proper comment.\n// MyMethod is deprecated.\n//\n// Deprecated: This may be removed in a future version.\n",
+			deprecated: true,
+		},
+		{
+			in:         "Deprecated: this is a proper deprecation notice.",
+			want:       "// MyMethod is deprecated.\n//\n// Deprecated: this is a proper deprecation notice.\n",
+			deprecated: true,
+		},
+		{
+			in:         "",
+			want:       "// MyMethod is deprecated.\n//\n// Deprecated: This may be removed in a future version.\n",
```

Comment:
This is the exact case I think we should include the "MyMethod is deprecated" statement.